src/pybind/mgr/progress/module.py

```diff
@@ -151,8 +151,14 @@ def pg_update(self, pg_dump, log):
             self._original_bytes_recovered = {}
             for pg in self._pgs:
                 pg_str = str(pg)
-                self._original_bytes_recovered[pg] = \
-                    pg_to_state[pg_str]['stat_sum']['num_bytes_recovered']
+                try:
+                    self._original_bytes_recovered[pg] = \
+                        pg_to_state[pg_str]['stat_sum']['num_bytes_recovered']
+                except KeyError:
+                    log.info('got KeyError, see http://tracker.ceph.com/issues/38157')
+                    log.info('pg_to_state as string: %s' % pg_to_state)
```

Comment:
instead of `str`, I'd call `repr()` and add 

```python
    def __repr__(self):
        return "PgId({},{})".format(self.pool_id, self.ps)
``` 
to `class PgId(object):`

Reply:
like so?

Reply:
lgtm